### PR TITLE
Fix heart disease notebook model evaluation, pipeline workflow, and README link

### LINKED ISSUE #1310: Various bugs in heart_prediction model.
**Describe the bug**
There are several bugs in this model:
1.The model instantiation and score calculation : lr = LogisticRegression(solver='liblinear') and score=cross_val_score(lr,x,y)  uses unscaled parameter (x) due to which the logistic regression model is stopping before it could find the best possible mathematical solution(or it is not converging ) .I tried to increase the iterations to 1000 and used liblinear solver but still it is not converging.
2. The UCI heart disease kaggle dataset link given in the readme is errornous.
3. The train_test_split imported in cell 114 is never actually used.
4. This is not a bug but the accuracy of the model can be improved by incorporating gridsearchcv.
**To Reproduce**
Steps to reproduce the behavior
Run the cell below 128th cell containing the code:lr = LogisticRegression()
score=cross_val_score(lr,x,y) 

**Solution**:
1. Scale x through standardscaler.fittransform() method and used the scaled variable instead in cross entropy calculation.
2. FIx the readme file error .
3. Inlcude gridsearchcv for logistic regression or decision tress models.

**Screenshots**

Image

**Desktop (please complete the following information):**
I ran the code on kaggle.

I am a gssoc "26 ai/agents_track contributor and would like to work on all these mentioned issues  and furthur increase the accuracy of this model.Please assign this issue to me under GSSOC '26.
Thankyou so much.

### PR DESCRIPTION
**Pull request by Lost-glitched**
Fixes  #1310
- Updated HeartDisease.ipynb to replace the broken `LogisticRegression` cross-validation block with a proper `Pipeline` + `GridSearchCV` evaluation
- Removed unused `train_test_split` import and added necessary `GridSearchCV`/`Pipeline` imports
- Preserved notebook outputs and execution state while applying source-only changes
- Corrected Readme.md dataset URL to the Kaggle Heart Disease UCI dataset

### CODE REVIEW

**Review by Sibam-Paul — APPROVED** [review]
looks good
